datafusion/scheduler/src/task.rs

```diff
@@ -203,15 +209,18 @@ impl Task {
 ///
 /// Dropping this will cancel the inflight query
 pub struct QueryResults {
-    inner: mpsc::UnboundedReceiver<Option<Result<RecordBatch>>>,
+    inner: mpsc::UnboundedReceiver<Option<ArrowResult<RecordBatch>>>,
+
+    /// The output schema of thi
+    schema: SchemaRef,
 
     /// Keep a reference to the [`QueryTask`] so it isn't dropped early
     #[allow(unused)]
     query: Arc<QueryTask>,
 }
 
 impl Stream for QueryResults {
-    type Item = Result<RecordBatch>;
+    type Item = ArrowResult<RecordBatch>;
```

Comment:
This is one quirk I was hoping to take the opportunity to fix, that DataFusion streams don't return a DataFusion error... It isn't the end of the world, but I'm curious how important this aspect is?